<launch>

  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="model" default="$(find haas_vf2)/urdf/haas_vf2.urdf"/>

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />

  <rosparam command="load" file="$(find haas_vf2)/config/base_to_y_axis_controller.yml" ns="base_to_y_axis_controller" />
  <rosparam command="load" file="$(find haas_vf2)/config/y_axis_to_x_axis_controller.yml" ns="y_axis_to_x_axis_controller" />
  <!-- <rosparam command="load" file="$(find haas_vf2)/config/gazebo_ros_control_params.yaml" /> -->

  <node name="vf2_controller_spawner" pkg="controller_manager" type="spawner" args="base_to_y_axis_controller y_axis_to_x_axis_controller"/>

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <!-- <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" args="-z 1.0 -unpause -urdf -model robot -param robot_description" respawn="false" output="screen" /> -->
  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" args="-z 18.5 -x 0.0 -y 0.0 -urdf -model robot -param robot_description" respawn="false" output="screen" />

  <node pkg="robot_state_publisher" type="robot_state_publisher"  name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node>

  <node name="vf2_position_server" pkg="haas_vf2" type="vf2_position_server.py" output="screen"/>
  <node name="vf2_motion_server" pkg="haas_vf2" type="vf2_motion_server.py" output="screen"/>

</launch>